src/flags/blocks.rs

```diff
@@ -23,72 +80,53 @@ impl Blocks {
-    /// # Errors
-    ///
-    /// This errors if any of the [ArgMatches] parameter arguments causes [Block]'s implementation
-    /// of [TryFrom::try_from] to return an [Err].
-    pub fn configure_from(matches: &ArgMatches, config: &Config) -> Result<Self, Error> {
-        let mut result: Result<Self, Error> = if matches.is_present("long") {
-            Ok(Self::long())
+    fn configure_from(matches: &ArgMatches, config: &Config) -> Self {
+        let mut blocks = if matches.is_present("long") {
+            Self::long()
         } else {
-            Ok(Default::default())
+            Default::default()
         };
 
         if matches.is_present("long") && !matches.is_present("ignore-config") {
             if let Some(value) = Self::from_config(config) {
-                result = Ok(value);
+                blocks = value;
             }
         }
 
         if let Some(value) = Self::from_arg_matches(matches) {
-            result = value;
+            blocks = value;
         }
 
         if matches.is_present("context") {
-            if let Ok(blocks) = result.as_mut() {
-                blocks.optional_insert_context();
-            }
+            blocks.optional_insert_context();
         }
-
         if matches.is_present("inode") {
-            if let Ok(blocks) = result.as_mut() {
-                blocks.optional_prepend_inode();
-            }
+            blocks.optional_prepend_inode();
         }
 
-        result
+        blocks
     }
 
     /// Get a potential `Blocks` struct from [ArgMatches].
     ///
     /// If the "blocks" argument is passed, then this returns a `Blocks` containing the parameter
     /// values in a [Some]. Otherwise if the "long" argument is passed, this returns
     /// [Blocks::long]. Finally if none of the previous happened, this returns [None].
-    ///
-    /// # Errors
-    ///
-    /// This errors if any of the parameter arguments causes [Block]'s implementation of
-    /// [TryFrom::try_from] to return an [Err].
-    fn from_arg_matches(matches: &ArgMatches) -> Option<Result<Self, Error>> {
+    fn from_arg_matches(matches: &ArgMatches) -> Option<Self> {
         if matches.occurrences_of("blocks") == 0 {
             return None;
         }
 
         if let Some(values) = matches.values_of("blocks") {
             let mut blocks: Vec<Block> = Vec::with_capacity(values.len());
             for value in values {
-                match Block::try_from(value) {
-                    Ok(block) => blocks.push(block),
-                    Err(message) => {
-                        return Some(Err(Error::with_description(
-                            &message,
-                            ErrorKind::ValueValidation,
-                        )))
-                    }
-                }
+                blocks.push(Block::try_from(value).unwrap_or_else(|_| {
+                    unreachable!(
+                        "Invalid value '{value}' for 'blocks' flag should be handled by `clap`"
```

Comment:
This is also present in the previous PR, but I was re-evaluating if we should mention clap?
How about just `Invalid value '{value}' for 'blocks'` ?

Reply:
I think we should make it clear why this statement should be unreachable because it is not obvious for new contributors. So, `clap` or `ArgMatchs` should be mentioned. 
I think it's fine to mention them in either the comment or in the panic message

Reply:
This message would be something that the users would see (if we mess something up). What I was thinking was that we could skip mentioning clap in the message but have it in a comment right next to it?

Reply:
Yeah. That's good. I'll change this and make another PR to change others `unreachable`